joefitzgerald/go-plus
# Consolidate Downstream Packages Into go-plus

The verdict is in: the experiment of decomposing go-plus into a collection of loosely coupled packages collaborating via services was not successful. It resulted in multiple levels of indirection, hampering the ability for people to:

* report issues
* find package settings
* understand the code base
* contribute to the package

This PR collapses the world back down into a single package: go-plus. It preserves the services that are provided for `go-config` and `go-get`, but I strongly encourage package authors making use of those packages to stop using `package-deps` to install `go-plus` (or the old `go-config` and `go-get` packages); it introduces a brittle package ecosystem, when changes like this are made.


* [x] Activate package via the `core:loaded-shell-environment` `activationHook` to ensure a patched environment is available for users of OS X / Linux in some launch scenarios (e.g. Dock / Unity Launcher)
* [x] Allow the go-plus panel to have `vertical` or `horizontal` orientations
* [x] Consolidate package configuration and arrange it into related sections
* [x] Consolidate `go-config`
* [x] Consolidate `go-get`
* [x] Consolidate `builder-go`
* [x] Consolidate `tester-go`
* [x] Consolidate `gofmt`
* [x] Consolidate `autocomplete-go`
* [x] Consolidate `gometalinter-linter`
* [x] Consolidate `navigator-go`
* [x] Work with @zmb3 to consolidate [`godoc`](https://github.com/zmb3/godoc)
* [x] Work with @zmb3 to consolidate [`gorename`](https://github.com/zmb3/gorename)
* [x] Work with @zheng1 to consolidate [`go-hyperclick`](https://github.com/zheng1/go-hyperclick)
* [x] Ensure `gometalinter --install` is run after `golang:update-tools`
* [x] Add a timeout to executor, to prevent runaway processes
* [x] Add configurable timeout for tests
* [x] Add configurable timeout for go get
* [x] Switch default for go to definition from godef to guru

- Fixes #487 
- Fixes #488 
- Fixes #491 
- Fixes #492 
- Fixes #495 
- Fixes #497 
- Fixes #498 
- Fixes https://github.com/joefitzgerald/tester-go/pull/36

<hr />

The following additional features will be added in subsequent PRs:

* Add option for go to definition that makes use of gogetdoc
* Add ability for the hyperclick provider to display godoc as an overlay on hover
* Allow coverage display style to be one of: `gutter`, `highlight`, `both`, adopting code from https://github.com/joefitzgerald/tester-go/pull/36
